src/inmanta/env.py

```diff
@@ -58,7 +67,53 @@ class ProcessEnv:
     Class to represent the Python environment this process is running in.
     """
 
-    env_path: str = sys.executable
+    python_path: str = sys.executable
+
+    @classmethod
+    def get_site_packages_dir(cls) -> str:
+        env_dir: str = os.path.dirname(os.path.dirname(cls.python_path))
+        return (
+            os.path.join(env_dir, "Lib", "site-packages")
+            if sys.platform == "win32"
+            else os.path.join(env_dir, "lib", "python%s" % sys.version[:3], "site-packages")
+        )
+
+    @classmethod
+    def install_from_index(
+        cls, requirements: List[Requirement], index_urls: Optional[List[str]] = None, upgrade: bool = False
+    ) -> None:
+        index_args: List[str] = (
+            []
+            if index_urls is None
+            else ["--index-url", index_urls[0], *chain.from_iterable(["--extra-index-url", url] for url in index_urls[1:])]
+            if index_urls
+            else ["--no-index"]
+        )
+        try:
+            process: subprocess.CompletedProcess = subprocess.run(
+                [
+                    cls.python_path,
+                    "-m",
+                    "pip",
+                    "install",
+                    *(["--upgrade"] if upgrade else []),
+                    *(str(requirement) for requirement in requirements),
+                    *index_args,
+                ],
+                stderr=subprocess.PIPE,
+            )
+            process.check_returncode()
+            # TODO: other install methods patch pkg_resources.working_set, do we need to do so here as well?
```

Comment:
yes (and then some for the namespaces, I think you are around made a method for that)

Reply:
Coincidentally I'm using `pkg_resources.working_set` to resolve another review comment, so I've got some idea now.

Reply:
(done)